### PR TITLE
Duplication Guard for AddApptCommandParser and DelApptCommandParser

### LINKED ISSUE #115: Bug Fix for NRIC duplication issue for addAppt
By entering the NRIC, i/s1234567A and i/s1234567B, it will accept the last
NRIC instead of rejecting it.

### PR DESCRIPTION
**Pull request by yadobler**
Fixes #115 as well as for DelApptCommandParser

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/AddApptCommandParser.java [hunk 1]
```diff
@@ -28,8 +28,8 @@ public class AddApptCommandParser implements Parser<AddApptCommand> {
      */
     public AddApptCommand parse(String args) throws ParseException {
         requireNonNull(args);
-        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_NRIC, PREFIX_DATE,
-                                                                  PREFIX_TIMEPERIOD);
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_NRIC, PREFIX_DATE, PREFIX_TIMEPERIOD);
+        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_NRIC, PREFIX_DATE, PREFIX_TIMEPERIOD);
```
Comment (yong-zhuo) [inline src/main/java/seedu/address/logic/parser/AddApptCommandParser.java:32]: LGTM, `verifyNoDuplicatePrefixesFor` is used well

File: src/test/java/seedu/address/logic/parser/AddApptCommandParserTest.java [hunk 2]
```diff
@@ -89,4 +91,16 @@ public void parse_compulsoryFieldsMissing_failure() {
...
+    public void parse_duplicateValues_throwsParserException() {
+        assertParseFailure(parser, NRIC_DESC_AMY + APPOINTMENT_DESC_AMY + NRIC_DESC_AMY,
+                           MESSAGE_DUPLICATE_FIELDS + PREFIX_NRIC);
+        assertParseFailure(parser, APPOINTMENT_DESC_AMY + " " + PREFIX_DATE + VALID_APPOINTMENT_DATE_DENTAL + " "
+                                   + NRIC_DESC_AMY, MESSAGE_DUPLICATE_FIELDS + PREFIX_DATE);
+        assertParseFailure(parser, APPOINTMENT_DESC_AMY + " " + PREFIX_TIMEPERIOD + VALID_APPOINTMENT_TIMEPERIOD_DENTAL
+                                   + " " + NRIC_DESC_AMY, MESSAGE_DUPLICATE_FIELDS + PREFIX_TIMEPERIOD);
+        assertParseFailure(parser, APPOINTMENT_DESC_BOB + " " + APPOINTMENT_DESC_AMY + NRIC_DESC_AMY,
+                           MESSAGE_DUPLICATE_FIELDS + PREFIX_DATE + " " + PREFIX_TIMEPERIOD);
+    }
```
Comment (yong-zhuo) [inline src/test/java/seedu/address/logic/parser/AddApptCommandParserTest.java:105]: Relevant test cases for the new addition 👍🏻

**Review by yong-zhuo — APPROVED** [review]
Overall, LGTM

**Review by PresleyChew — APPROVED** [review]
LGTM

**Review by hweinian — COMMENTED** [review]
lgtm

**Pull request merged**